metagraph/src/graph/graph_extensions/node_rc.cpp

```diff
@@ -58,13 +58,29 @@ NodeRC::NodeRC(const DeBruijnGraph &graph, bool construct_index) : graph_(&graph
         auto it = rc_edges.begin();
         for (size_t i = 0; i < path.size(); ++i, ++it) {
             if (*it) {
-                std::lock_guard<std::mutex> lock(mu);
-                rc_nodes_prefix.emplace_back(path[i]);
+                bool found = false;
+                boss.call_outgoing(*it, [&](edge_index adjacent_edge) {
+                    found |= (dbg_succ->boss_to_kmer_index(adjacent_edge) != DeBruijnGraph::npos);
```

Comment:
This will only work with an edge mask in dbg_succinc. You actually need to check if the last character is not dummy